bip-0340.mediawiki

```diff
@@ -141,40 +142,46 @@ The algorithm ''PubKey(sk)'' is defined as:
 
 Note that we use a very different public key format (32 bytes) than the ones used by existing systems (which typically use elliptic curve points as public keys, or 33-byte or 65-byte encodings of them). A side effect is that ''PubKey(sk) = PubKey(bytes(n - int(sk))'', so every public key has two corresponding secret keys.
 
+==== Public Key Conversion ====
+
 As an alternative to generating keys randomly, it is also possible and safe to repurpose existing key generation algorithms for ECDSA in a compatible way. The secret keys constructed by such an algorithm can be used as ''sk'' directly. The public keys constructed by such an algorithm (assuming they use the 33-byte compressed encoding) need to be converted by dropping the first byte. Specifically, [[bip-0032.mediawiki|BIP32]] and schemes built on top of it remain usable.
 
 ==== Default Signing ====
 
 Input:
 * The secret key ''sk'': a 32-byte array
 * The message ''m'': a 32-byte array
+* Auxiliary random data ''a'': a byte array of length 0 to 32 (inclusive)
```

Comment:
How about we fix `a` to be 32 bytes (or alternatively 0 or 32 bytes). This would make the test vectors much simpler because otherwise we will want to have a test vector with an `a` that is in between. Making the libsecp compatible with variable len `a`  is unnecessarily complicated as the caller would need to provide a byte array for `a` that also encodes the length. So either we make `a` (0 or 32 bytes) or we need to design the test vectors such that only one of them has an in-between `a` and skip that one it in libsecp.

Reply:
this is resolved